src/pokeradar.c

```diff
@@ -0,0 +1,590 @@
+#include <nitro.h>
+#include <string.h>
+
+#include "inlines.h"
+
+#include "struct_decls/struct_020508D4_decl.h"
+#include "overlay101/struct_ov101_021D5D90_decl.h"
+
+#include "struct_defs/struct_0201CFEC.h"
+#include "struct_defs/struct_0203CDB0.h"
+#include "struct_defs/struct_020698E4.h"
+
+#include "unk_02005474.h"
+#include "heap.h"
+#include "unk_0201CED8.h"
+#include "unk_0202D7A8.h"
+#include "unk_02039C80.h"
+#include "unk_0203E880.h"
+#include "unk_020508D4.h"
+#include "unk_02054D00.h"
+#include "unk_020553DC.h"
+#include "unk_0205DAC8.h"
+#include "unk_0205E7D0.h"
+#include "unk_02061804.h"
+#include "pokeradar.h"
+#include "unk_020711EC.h"
+#include "unk_0207D3B8.h"
+#include "overlay005/ov5_021F2D20.h"
+#include "overlay006/ov6_022430C4.h"
+
+#define NUM_GRASS_PATCHES 4
+#define RADAR_BATTERY_STEPS 50
+
+typedef struct {
+    int unk_00;
+    int unk_04;
+    int shakeType;
+    BOOL active;
+    BOOL continueChain;
+    BOOL shiny;
+    UnkStruct_ov101_021D5D90 * unk_18;
+    VecFx32 position;
+} GrassPatch;
+
+typedef struct RadarChain_t {
+    int shakeType;
+    int count;
+    int species;
+    int level;
+    BOOL active;
+    BOOL unk_14;
+    BOOL unk_18;
+    GrassPatch patch[NUM_GRASS_PATCHES];
+    UnkStruct_0201CFEC unk_BC;
+    u8 unk_D0;
+} RadarChain;
+
+static BOOL CheckTileIsGrass(UnkStruct_0203CDB0 * param0, const fx32 param1, const int param2, const int param3, const u8 param4, const u8 param5, GrassPatch * patch);
+static BOOL sub_020698AC(const RadarChain * chain, const int param1, const int param2, u8 * param3);
+static void sub_020698E4(UnkStruct_0203CDB0 * param0, RadarChain * chain);
+static u8 sub_0206994C(UnkStruct_0203CDB0 * param0);
+static BOOL CheckPatchContinueChain(const u8 patchRing, const int battleResult);
+static BOOL CheckPatchShiny(const int param0);
+static void IncWithCap(int * param0);
+
+RadarChain * CreateRadarChain (const int param0) {
+    RadarChain * chain;
+
+    chain = Heap_AllocFromHeap(param0, sizeof(RadarChain));
```

Comment:
We use C99; prefer declare + define on the same line wherever possible, so long as it matches.

```suggestion
    RadarChain *chain = Heap_AllocFromHeap(param0, sizeof(RadarChain));
```